src/docs/release/breaking-changes/rendereditable-stop-painting-invalid-selection.md

```diff
@@ -0,0 +1,96 @@
+---
+title: Change RenderEditable to stop painting the caret when its selection is invalid
+description: RenderEditable no longer paints the caret when its selection is (-1, -1).
+---
+
+## Summary
+
+When a text field's selection is `TextSelection.collapsed(offset: -1)`,
+`RenderEditable` no longer paints the caret, nor will its `onCaretChanged` 
```

Comment:
If the selection changes from valid to invalid, or vice versa, will we notify the ‘TextEditingController’?

Reply:
No it does not (you meant the `onCaretChanged` callback right?) when the caret becomes invisible. But IIRC it does when the caret becomes visible again. To fix this we would need another breaking change (to change the callback's signature).

Reply:
I have plans to change the "showCaretOnScreen" thing to not rely on the "onCaretChanged" callback because it's not reliable (partially because of this ^).